gwcs/api.py

```diff
@@ -0,0 +1,212 @@
+        """
+        result = self.invert(*world_arrays, with_units=False)
+        return result
+    
+    
+    def world_to_array_index_values(self, *world_arrays):
+        """
+        Convert world coordinates to array indices.
+        This is the same as `~BaseLowLevelWCS.world_to_pixel_values` except that
+        the indices should be returned in ``(i, j)`` order, where for an image
+        ``i`` is the row and ``j`` is the column (i.e. the opposite order to
+        `~BaseLowLevelWCS.pixel_to_world_values`). The indices should be
+        returned as rounded integers.
+        """
+        result = self.invert(*world_arrays, with_units=False)[::-1]
+        return result # astype(int)
+
+    @property
+    def array_shape(self):
+        """
+        The shape of the data that the WCS applies to as a tuple of
+        length `~BaseLowLevelWCS.pixel_n_dim`.
+        If the WCS is valid in the context of a dataset with a particular
+        shape, then this property can be used to store the shape of the
+        data. This can be used for example if implementing slicing of WCS
+        objects. This is an optional property, and it should return `None`
+        if a shape is not known or relevant.
+        The shape should be given in ``(row, column)`` order (the convention
+        for arrays in Python).
+        """
+        return self._array_shape
+
+    @array_shape.setter
+    def array_shape(self, value):
+        self._array_shape = value
+        
+    @property
+    def pixel_bounds(self):
+        """
+        The bounds (in pixel coordinates) inside which the WCS is defined,
+        as a list with `~BaseLowLevelWCS.pixel_n_dim` ``(min, max)`` tuples.
+        The bounds should be given in ``[(xmin, xmax), (ymin, ymax)]``
+        order. WCS solutions are sometimes only guaranteed to be accurate
+        within a certain range of pixel values, for example when defining a
+        WCS that includes fitted distortions. This is an optional property,
+        and it should return `None` if a shape is not known or relevant.
+        """
+        return self.bounding_box
+
+    @property
+    def axis_correlation_matrix(self):
+        """
+        Returns an (`~BaseLowLevelWCS.world_n_dim`,
+        `~BaseLowLevelWCS.pixel_n_dim`) matrix that indicates using booleans
+        whether a given world coordinate depends on a given pixel coordinate.
+        This defaults to a matrix where all elements are `True` in the absence of
+        any further information. For completely independent axes, the diagonal
+        would be `True` and all other entries `False`.
+        """
+
+        return separable.separability_matrix(self.forward_transform)
+
+    def serialized_classes(self):
+        """
+        Indicates whether Python objects are given in serialized form or as
+        actual Python objects.
+        """
+        return False
+
+    def world_axis_object_classes(self):
+        raise NotImplementedError()
```

Comment:
Is the plan to implement these to facilitate a common high level API object? (In addition to gWCS implementing the high level api?)